eos-split-flatpak-repo

```diff
@@ -114,47 +114,69 @@ def gather_remotes(repo):
     return os_remotes, flatpak_remotes
 
 
+def get_refspec_repo_type(refspec, os_remotes, flatpak_remotes):
+    """Determine the repo type this refspec should be a part of
+
+    Returns a RepoType or None if it can't be determined.
+    """
+    _, remote, ref = OSTree.parse_refspec(refspec)
+    if remote:
+        if remote in os_remotes:
+            return RepoType.OS
+        elif remote in flatpak_remotes:
+            return RepoType.Flatpak
+        else:
+            # The remote has been deleted, so fall through to the
+            # ref heuristics.
+            logger.warning('Refspec %s remote %s no longer exists',
+                           refspec, remote)
+
+    ref_parts = ref.split('/')
+    if len(ref_parts) > 1:
+        ref_prefix = ref_parts[0]
+    else:
+        ref_prefix = ''
+
+    if ref_prefix in ('app', 'runtime', 'deploy'):
+        return RepoType.Flatpak
+    elif ref_prefix in ('os', 'ostree'):
+        return RepoType.OS
+    elif ref_prefix in ('appstream', 'appstream2'):
+        if remote:
+            # This came from a deleted flatpak remote
+            return RepoType.Flatpak
+        else:
+            # Ignore locally generated appstream refs in case someone
+            # ran flatpak build-update-repo.
+            logger.debug('Ignoring generated ref: %s', ref)
+            return None
+    elif ref == 'ostree-metadata':
+        # Even if this came from a remote we don't know what type it
+        # was, so just ignore it.
+        logger.debug('Ignoring generated ref: %s', ref)
+        return None
+    else:
+        # XXX: Should this fail here?
```

Comment:
I think it’s fair to warn and return `None` so the ref is kept in both new repos, as you are doing.